pennylane/data/data_manager/params.py

```diff
@@ -126,9 +126,36 @@ def format_param_args(param: ParamName, details: Any) -> Union[ParamArg, list[Pa
     return details
 
 
-def format_params(**params: Any) -> dict[ParamName, Union[ParamArg, ParamVal]]:
-    """Converts params to a dictionary whose keys are parameter names and
-    whose values are single ``ParamaterArg`` objects or lists of parameter values."""
-    return {
+def format_params(**params: Any) -> list[dict[str:ParamName, str : Union[ParamArg, ParamVal]]]:
+    """Converts params to a list of dictionaries whose values are parameter names and
+    single ``ParamaterArg`` objects or lists of parameter values."""
+
+    input_params = {
         param_name: format_param_args(param_name, param) for param_name, param in params.items()
     }
+    return [{"name": k, "values": v} for k, v in input_params.items()]
+
+
+def provide_defaults(
+    data_name: str, params: list[dict[str:ParamName, str : Union[ParamArg, ParamVal]]]
+) -> list[dict[str:ParamName, str : Union[ParamArg, ParamVal]]]:
+    """
+    Provides default parameters to the qchem and qspin query parameters if the parameter
+    names are missing from the provided ``params``.
+    """
+    param_names = [param["name"] for param in params]
+    if data_name == "qchem":
```

Comment:
Is this needed? The API should handle empty parameters on its own

Reply:
Yes the API would provide all datasets for a given parameter if a value wasn't specified so if we want to provide defaults in the absence of specified parameters e.g `qml.data.load("qchem", molname="H2")` we need to explicitly provide the `{"default": true}` for the remaining parameters. 

Conversely, this behaviour is leveraged for the `qml.data.load("qchem", molname="H2", "bondlength"='full')` case where the full set of datasets for all bondlengths is retrieved by simply not passing in a parameter value for it.